---
title: "Preliminary MSH endophytes markdown"
output:
  html_document:
    df_print: paged
editor_options:
  chunk_output_type: console
---

Preliminary data analysis of Sanger sequencing data from cultures isolated from Mount St. Helens (MSH) leaf samples collected July-Sept 2017 by ERW. Samples were surface-sterilized and either plated on malt extract agar plates or processed for storage at -80C (for direct extraction). Analysis is being done for a poster for Alumni Night 2019.

MAFFT v 7.427 on CIPRES

#convert to phylip from fasta: http://phylogeny.lirmm.fr/phylo_cgi/data_converter.cgi

Analysis in mothur:
```{mothur}
#version 1.42.3
summary.seqs(fasta=trimmed_.fasta)
unique.seqs(fasta=current)
count.seqs(name=trimmed_.names, group=leafID4.txt)

cluster(phylip=mafft.dist, method=furthest, cutoff=0.01)
classify.seqs(fasta=trimmed_.unique.fasta,count=trimmed_.count_table, template=UNITEv8_sh_97_s.fasta,taxonomy=UNITEv8_sh_97_s.tax,cutoff=80)
classify.otu(list=mafft.fn.01.list, taxonomy=trimmed_.unique.UNITEv8_sh_97_s.wang.taxonomy,count=trimmed_.count_table,label=0.01,cutoff=80,basis=otu,probs=T)

#adjusted for direction as a setting in MAFFT
cluster(phylip=mafft.adj.dist, method=furthest, cutoff=0.01)
make.shared(list=mafft.adj.fn.list, group=leafID4.txt, label=0.01)
classify.otu(list=mafft.adj.fn.list, taxonomy=trimmed_.unique.UNITEv8_sh_97_s.wang.taxonomy,count=trimmed_.count_table,label=0.01,cutoff=80,basis=otu,probs=T)

```

Load libraries:
```{r load_libraries}
library(vegan)
library(ggplot2)
library(indicspecies)
library(car)
library(phyloseq) #if not installed already, should be done through Bioconductor

#install gg_ordiplots from GitHub if not already installed
#install.packages("remotes")
#remotes::install_github("jfq3/ggordiplots")
library(ggordiplots)


```


Now import the data:
```{r import}
#setwd("C:\\Users\\emyle\\Dropbox\\School\\Data\\Projects\\MSH\\MSH_endophytes")

PP_com <- read.csv("adj.com.csv", row.names = 1)
head(PP_com[,1:5])

env_all <- read.csv("adj_env.csv")
head(env_all)

PP_PA <- read.csv("adj_PA_com.csv",row.names=1)
head(PP_PA[1:5,1:5])

PP_a <- read.csv("PP_abund.csv",row.names=1)
head(PP_a[1:5,1:5])

env <- read.csv("adj_PP_env.csv", row.names = 1)
head(env)

#subset original community so that it is just SASI and ALVI, since they are the most abundant species
#SA_env <- subset(env, env$Species =="SASI" | env$Species == "ALVI")
#SA_com<- PP_com[rownames(SA_env),]

#SA_com[1:5,1:5]
#SA_env[1:5,]

#use the PA community instead because it's been adjusted for the inflated abundances
#subset original community so that it is just SASI and ALVI, since they are the most abundant species
SA_env <- subset(env, env$Species =="SASI" | env$Species == "ALVI")
SA_com<- PP_PA[rownames(SA_env),]

SA_com[1:5,1:5]
SA_env[1:5,]


D_env <- subset(env, env$Species =="SASI" | env$Species == "ALVI" | env$Species=="POTR")
D_com<- PP_PA[rownames(D_env),]

D_com[1:5,1:5]
D_env[1:5,]
```


```{r nmds_PA}
set.seed(226) #because it's permutative

#total com
#conifer vs deciduous com
#harvest effects
#SASI vs ALVI com
#interacting effects between tree type and harvest

df.mds<-metaMDS(PP_com, distance = "bray", k = 3, trymax = 50, autotransform=TRUE)
df.mds

#generate df of nmds scores for ggplot
data.scores = as.data.frame(scores(df.mds))
data.scores$Sample = env_all$Group
data.scores$Species = env_all$Species
data.scores$Site = env_all$Site_Cat
data.scores$Harvest = env_all$Harvest

#changes order from alphabetical to chronological
data.scores$Harvest <- relevel(data.scores$Harvest, "July")

head(data.scores)

xx = ggplot(data.scores, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 4, aes( shape =Species, colour = Harvest))+ 
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank()) + 
    labs(x = "NMDS1", colour = "Harvest", y = "NMDS2", shape = "Species")  + 
    scale_colour_manual(values = c("#009E73", "#E69F00","blue","red")) 
 
xx #entire Sanger-sequenced community, including NP and RW


PA.mds<-metaMDS(PP_PA,distance="bray",k=3,trymax = 50)
PA.mds

data.scores_PA = as.data.frame(scores(PA.mds))
data.scores_PA$Species = env$Species
data.scores_PA$Harvest = env$Harvest
#changes order from alphabetical to chronological
data.scores_PA$Harvest <- relevel(data.scores_PA$Harvest, "July")

head(data.scores_PA)

#get the hull info to plot on the ggplot object
p.sp <- gg_ordiplot(PA.mds, groups=env$Species,hull=TRUE,ellipse = T,kind="sd",conf = 0.95)
p.sp.hulls <- p.sp$df_hull
p.sp.ell <- p.sp$df_ellipse
p.h <- gg_ordiplot(PA.mds, groups=env$Harvest,hull=TRUE,ellipse = T,kind="sd",conf=0.95)
p.h.hulls<-p.h$df_hull
p.h.ell <- p.h$df_ellipse

PA.sp = ggplot(data.scores_PA, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 4, aes(color =Species, shape = Harvest))+ 
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank()) + 
    labs(x = "NMDS1", colour = "Species", y = "NMDS2", shape = "Harvest")  #+ 
    #scale_colour_manual(values = c("#8c510a","#d8b365","#f6e8c3","#c7eae5","#5ab4ac","#01665e")) 
 
jpeg(filename="PP_nmds_gg.jpeg", width=10, height=10, units="in", res = 300, pointsize = 20)
PA.sp + geom_path(data=p.sp.ell, aes(x=x,y=y,color=Group), show.legend=F) + scale_colour_manual(values = c("#8c510a","#d8b365","#f6e8c3","#c7eae5","#5ab4ac","#01665e"))

#+ geom_polygon(data=p.sp.ell, aes(x=x, y=y, color=Group,fill=Group),alpha=0.25,show.legend = FALSE) + scale_fill_manual(values = c("#8c510a","#d8b365","#f6e8c3","#c7eae5","#5ab4ac","#01665e")) + scale_colour_manual(values = c("#8c510a","#d8b365","#f6e8c3","#c7eae5","#5ab4ac","#01665e"))
dev.off()

#make one to look at harvest overlap
PA.h = ggplot(data.scores_PA, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 4, aes(color =Harvest, shape = Species))+ 
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank()) + 
    labs(x = "NMDS1", colour = "Harvest", y = "NMDS2", shape = "Species")  #+ 
    #scale_colour_manual(values = c("#8c510a","#d8b365","#f6e8c3","#c7eae5","#5ab4ac","#01665e")) 
 
jpeg(filename="PP_nmds_harvest_gg.jpeg", width=10, height=10, units="in", res = 300, pointsize = 20)
PA.h + geom_path(data=p.h.ell, aes(x=x, y=y, color=Group),show.legend = FALSE) + scale_colour_manual(values = c("#49006a","#ae017e","#f768a1"))
dev.off()

#PA.h + geom_polygon(data=p.h.hulls, aes(x=x, y=y, color=Group,fill=Group),alpha=0.25,show.legend = FALSE) + scale_fill_manual(values = c("#8c510a","#d8b365","#f6e8c3","#c7eae5","#5ab4ac","#01665e")) + scale_colour_manual(values = c("#8c510a","#d8b365","#f6e8c3","#c7eae5","#5ab4ac","#01665e"))
#ugly and uninformative
```


```{r nmds_D}
#now for the SASI, POTR, and ALVI only community
D.mds<-metaMDS(D_com, distance = "bray", k = 2, trymax = 50, autotransform=TRUE)
D.mds

data.scores_D = as.data.frame(scores(D.mds))
data.scores_D$Species = D_env$Species
data.scores_D$Harvest = D_env$Harvest
data.scores_D$Harvest <- relevel(data.scores_D$Harvest, "July")
 
head(data.scores_D)

#get the hull info to plot on the ggplot object
D_h <- gg_ordiplot(D.mds, groups=D_env$Harvest,hull=T,ellipse = T)
hhulls <- D_h$df_hull

#must exclude extra levels
D_env$Species <- factor(D_env$Species)
D_sp <- gg_ordiplot(D.mds, groups=D_env$Species,hull=T,ellipse = T,kind="sd",conf = 0.95)
shulls <- D_sp$df_hull
ssp <- D_sp$df_ellipse

DxD = ggplot(data.scores_D, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 4, aes(color =Species, shape = Harvest))+ 
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank()) + 
    labs(x = "NMDS1", colour = "Species", y = "NMDS2", shape = "Harvest")  
   # scale_colour_manual(values = c("#edf8b1","#7fcdbb","#2c7fb8")) 
 
jpeg(filename="D_nmds_gg.jpeg", width=10, height=10, units="in", res = 300, pointsize = 20)
DxD + geom_path(data=ssp, aes(x=x, y=y, color=Group),show.legend = FALSE) + scale_colour_manual(values = c("#C6D960","#7fcdbb","#2c7fb8"))
dev.off()

#by harvest
hD = ggplot(data.scores_D, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 4, aes(color =Harvest, shape = Species))+ 
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank()) + 
    labs(x = "NMDS1", colour = "Harvest", y = "NMDS2", shape = "Species") 

jpeg(filename="D_nmds_harvest_gg.jpeg", width=10, height=10, units="in", res = 300, pointsize = 20)
hD + geom_polygon(data=hhulls, aes(x=x, y=y, color=Group,fill=Group),alpha=0.25,show.legend = FALSE) + scale_fill_manual(values = c("#49006a","#ae017e","#f768a1")) + scale_colour_manual(values = c("#49006a","#ae017e","#f768a1"))
dev.off()
```


```{r nmds_SA}
#now for the SASI and ALVI only community
SA.mds<-metaMDS(SA_com, distance = "bray", k = 2, trymax = 50, autotransform=TRUE)
SA.mds

data.scores_sa = as.data.frame(scores(SA.mds))
data.scores_sa$Species = SA_env$Species
data.scores_sa$Harvest = SA_env$Harvest
data.scores_sa$Harvest <- relevel(data.scores_sa$Harvest, "July")
 
head(data.scores_sa)

#get the hull info to plot on the ggplot object
SA_h <- gg_ordiplot(SA.mds, groups=SA_env$Harvest,hull=T,ellipse = T)
hhulls <- SA_h$df_hull


SA_sp <- gg_ordiplot(SA.mds, groups=SA_env$Species,hull=TRUE,ellipse = T,kind="sd",conf = 0.95)
shulls <- SA_sp$df_hull
ssp <- SA_sp$df_ellipse

xsa = ggplot(data.scores_sa, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 4, aes(color =Species, shape = Harvest))+ 
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank()) + 
    labs(x = "NMDS1", colour = "Species", y = "NMDS2", shape = "Harvest")  
   # scale_colour_manual(values = c("#edf8b1","#7fcdbb","#2c7fb8")) 
 
jpeg(filename="SA_nmds_gg.jpeg", width=10, height=10, units="in", res = 300, pointsize = 20)
xsa + geom_path(data=ssp, aes(x=x, y=y, color=Group),show.legend = FALSE) + scale_colour_manual(values = c("#C6D960","#7fcdbb","#2c7fb8"))
dev.off()

#by harvest
hsa = ggplot(data.scores_sa, aes(x = NMDS1, y = NMDS2)) + 
    geom_point(size = 4, aes(color =Harvest, shape = Species))+ 
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank()) + 
    labs(x = "NMDS1", colour = "Harvest", y = "NMDS2", shape = "Species") 

jpeg(filename="SA_nmds_harvest_gg.jpeg", width=10, height=10, units="in", res = 300, pointsize = 20)
hsa + geom_polygon(data=hhulls, aes(x=x, y=y, color=Group,fill=Group),alpha=0.25,show.legend = FALSE) + scale_fill_manual(values = c("#49006a","#ae017e","#f768a1")) + scale_colour_manual(values = c("#49006a","#ae017e","#f768a1"))
dev.off()
```


```{r specaccum}
#now make the species accumulation curves
accurve<-specaccum(PP_com, method="random", permutations=100)
plot(accurve)

#species accumulation curve figure
Paccurve<-specaccum(PP_PA, method="random", permutations=999)
jpeg(filename="PP_specaccum3.jpeg", width=8, height=8, units="in", res = 300, pointsize = 20)
plot(Paccurve,xlab="Number of Leaves",ylab="Richness")
dev.off()

#for the SA communityy
saccurve<-specaccum(SA_com, method="random", permutations=999)
plot(saccurve,xlab="Number of Leaves",ylab="Richness")

#for the decid com
Daccurve<-specaccum(D_com, method="random", permutations=999)
plot(Daccurve,xlab="Number of Leaves",ylab="Richness")
```


```{r permanova}
#enter pairwise permanova function
pairwise.adonis <- function(x,factors, sim.method, p.adjust.m)
{
library(vegan)
co = as.matrix(combn(unique(factors),2))
pairs = c()
F.Model =c()
R2 = c()
p.value = c()

for(elem in 1:ncol(co)){
ad = adonis(x[factors %in% c(as.character(co[1,elem]),as.character(co[2,elem])),] ~ 
factors[factors %in% c(as.character(co[1,elem]),as.character(co[2,elem]))] , method =sim.method);
pairs = c(pairs,paste(co[1,elem],'vs',co[2,elem]));
F.Model =c(F.Model,ad$aov.tab[1,4]);
R2 = c(R2,ad$aov.tab[1,5]);
p.value = c(p.value,ad$aov.tab[1,6])
}
p.adjusted = p.adjust(p.value,method=p.adjust.m)
pairw.res = data.frame(pairs,F.Model,R2,p.value,p.adjusted)
return(pairw.res)
}


# #by species
# adonis(PP_com~Species,env, perm=999,method="bray")
# pairwise.adonis(PP_com,env$Species,sim.method="bray",p.adjust.m = "bonferroni")
# #df=5,80; F=1.6184, p=0.002
# 
# #by harvest
# adonis(PP_com~Harvest,env, perm=999,method="bray")
# pairwise.adonis(PP_com,env$Harvest,sim.method="bray",p.adjust.m = "bonferroni")
# #df=2,83; F=1.4347, p=0.078
# 
# #by tree type (conifer vs deciduous)
# adonis(PP_com~Type,env, perm=999,method="bray")
# #df=1,84; F=1.6142, p=0.074


#by species
adonis(PP_PA~Species,env, perm=999,method="bray")
# Call:
# adonis(formula = PP_PA ~ Species, data = env, permutations = 999,      method = "bray") 
# 
# Permutation: free
# Number of permutations: 999
# 
# Terms added sequentially (first to last)
# 
#            Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)   
# Species     5     3.423 0.68457  1.7781 0.07671  0.005 **
# Residuals 107    41.195 0.38500         0.92329          
# Total     112    44.618                 1.00000       
# # ---
# # Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

pairwise.adonis(PP_PA,env$Species,sim.method="bray",p.adjust.m = "bonferroni")
#           pairs   F.Model         R2 p.value p.adjusted
# 1  ALVI vs POTR 1.8258840 0.03523112   0.062      0.930
# 2  ALVI vs PSME 2.8953380 0.05088884   0.009      0.135
# 3  ALVI vs ABPR 0.8909311 0.02077202   0.415      1.000
# 4  ALVI vs SASI 3.1208370 0.03847146   0.009      0.135
# 5  ALVI vs TSHE 1.3610501 0.02814352   0.219      1.000
# 6  POTR vs PSME 1.1564982 0.04994271   0.258      1.000
# 7  POTR vs ABPR 1.3132513 0.11608080   0.393      1.000
# 8  POTR vs SASI 2.0043783 0.04175407   0.065      0.975
# 9  POTR vs TSHE 0.8575937 0.05408095   0.632      1.000
# 10 PSME vs ABPR 1.5048504 0.09705675   0.064      0.960
# 11 PSME vs SASI 2.1567742 0.04135176   0.059      0.885
# 12 PSME vs TSHE 0.4621736 0.02374728   0.948      1.000
# 13 ABPR vs SASI 0.8197368 0.02111649   0.616      1.000
# 14 ABPR vs TSHE 1.1666667 0.14285714   0.443      1.000
# 15 SASI vs TSHE 1.2853971 0.02902530   0.208      1.000


#by harvest
adonis(PP_PA~Harvest,env, perm=999,method="bray")
# Call:
# adonis(formula = PP_PA ~ Harvest, data = env, permutations = 999,      method = "bray") 
# 
# Permutation: free
# Number of permutations: 999
# 
# Terms added sequentially (first to last)
# 
#            Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  
# Harvest     2     1.153 0.57671  1.4595 0.02585  0.091 .
# Residuals 110    43.465 0.39514         0.97415         
# Total     112    44.618                 1.00000         
# ---
# Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

pairwise.adonis(PP_PA,env$Harvest,sim.method="bray",p.adjust.m = "bonferroni")
#                 pairs   F.Model         R2 p.value p.adjusted
# 1   September vs July 2.1748262 0.02614765   0.046      0.138
# 2 September vs August 1.1666072 0.01437299   0.275      0.825
# 3      July vs August 0.8812526 0.01471667   0.466      1.000


#by tree type (conifer vs deciduous)
adonis(PP_PA~Type,env, perm=999,method="bray")
# Call:
# adonis(formula = PP_PA ~ Type, data = env, permutations = 999,      method = "bray") 
# 
# Permutation: free
# Number of permutations: 999
# 
# Terms added sequentially (first to last)
# 
#            Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  
# Type        1     0.781 0.78054  1.9764 0.01749  0.035 *
# Residuals 111    43.838 0.39493         0.98251         
# Total     112    44.618                 1.00000         
# ---
# Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

#by type*harvest
adonis(PP_PA~Type*Harvest,env, perm=999,method="bray")
# Call:
# adonis(formula = PP_PA ~ Type * Harvest, data = env, permutations = 999,      method = "bray") 
# 
# Permutation: free
# Number of permutations: 999
# 
# Terms added sequentially (first to last)
# 
#               Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  
# Type           1     0.781 0.78054 1.98015 0.01749  0.044 *
# Harvest        2     1.120 0.55984 1.42026 0.02509  0.104  
# Type:Harvest   2     0.540 0.27017 0.68539 0.01211  0.855  
# Residuals    107    42.178 0.39418         0.94530         
# Total        112    44.618                 1.00000         
# ---
# Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

#check dispersion
PAdist <- vegdist(PP_PA, method="bray")
PAmodh <- betadisper(PAdist,env$Harvest)
PAmodh
anova(PAmodh)
permutest(PAmodh,pairwise=TRUE,permutations=999)
PAmodh.HSD <- TukeyHSD(PAmodh)
plot(PAmodh.HSD)

#harvest
# Analysis of Variance Table
# 
# Response: Distances
#            Df  Sum Sq  Mean Sq F value Pr(>F)
# Groups      2  0.2109 0.105433  1.1392 0.3238
# Residuals 110 10.1809 0.092553 

PAmod <- betadisper(PAdist,env$Species)
PAmod
anova(PAmod)
permutest(PAmod,pairwise=TRUE,permutations=999)
PAmod.HSD <- TukeyHSD(PAmod)
plot(PAmod.HSD)

#species
# Analysis of Variance Table
# 
# Response: Distances
#            Df  Sum Sq Mean Sq F value Pr(>F)
# Groups      5  0.9481 0.18961   1.759 0.1275
# Residuals 107 11.5339 0.10779

#p>0.05 for both so PERMANOVAs are OKAY to interpret, and dispersion is similiar within vs between groups
```


```{r permanova_SA}
#differences between SASI and ALVI communities
adonis(SA_com~Species,SA_env, perm=999,method="bray")
# Call:
# adonis(formula = SA_com ~ Species, data = SA_env, permutations = 999,      method = "bray") 
# 
# Permutation: free
# Number of permutations: 999
# 
# Terms added sequentially (first to last)
# 
#           Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)   
# Species    1    1.1278 1.12783  3.1208 0.03847  0.003 **
# Residuals 78   28.1881 0.36139         0.96153          
# Total     79   29.3160                 1.00000          
# ---
# Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

adonis(SA_com~Harvest,SA_env, perm=999,method="bray")
# Call:
# adonis(formula = SA_com ~ Harvest, data = SA_env, permutations = 999,      method = "bray") 
# 
# Permutation: free
# Number of permutations: 999
# 
# Terms added sequentially (first to last)
# 
#           Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  
# Harvest    2    1.2013 0.60066  1.6451 0.04098  0.052 .
# Residuals 77   28.1147 0.36513         0.95902         
# Total     79   29.3160                 1.00000         
# ---
# Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

SAdist <- vegdist(SA_com, method="bray")
SAmodh <- betadisper(SAdist,SA_env$Harvest)
SAmodh
anova(SAmodh)
permutest(SAmodh,pairwise=TRUE,permutations=999)
SAmodh.HSD <- TukeyHSD(SAmodh)
plot(SAmodh.HSD)

#harvest
# Analysis of Variance Table
# 
# Response: Distances
#           Df Sum Sq Mean Sq F value Pr(>F)
# Groups     2 0.3883 0.19415  1.5288 0.2233
# Residuals 77 9.7786 0.12699                 

SAmod <- betadisper(SAdist,SA_env$Harvest)
SAmod
anova(SAmod)
permutest(SAmod,pairwise=TRUE,permutations=999)
SAmod.HSD <- TukeyHSD(SAmod)
plot(SAmod.HSD)

#species
# Analysis of Variance Table
# 
# Response: Distances
#           Df Sum Sq Mean Sq F value Pr(>F)
# Groups     2 0.3883 0.19415  1.5288 0.2233
# Residuals 77 9.7786 0.12699  

###same here, p<0.05

#for decid coms
adonis(D_com~Species,D_env, perm=999,method="bray")
# Call:
# adonis(formula = D_com ~ Species, data = D_env, permutations = 999,      method = "bray") 
# 
# Permutation: free
# Number of permutations: 999
# 
# Terms added sequentially (first to last)
# 
#           Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)   
# Species    2     1.807 0.90367  2.4337 0.05298  0.004 **
# Residuals 87    32.305 0.37132         0.94702          
# Total     89    34.112                 1.00000          
# ---
# Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

adonis(D_com~Harvest,D_env, perm=999,method="bray")
# Call:
# adonis(formula = D_com ~ Harvest, data = D_env, permutations = 999,      method = "bray") 
# 
# Permutation: free
# Number of permutations: 999
# 
# Terms added sequentially (first to last)
# 
#           Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  
# Harvest    2     1.102 0.55104  1.4523 0.03231  0.088 .
# Residuals 87    33.010 0.37942         0.96769         
# Total     89    34.112                 1.00000         
# ---
# Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

```


```{r div_PA}
env$Shannon <- diversity(PP_PA)
mean(env$Shannon);sd(env$Shannon)
# [1] 0.1656192
# [1] 0.3110119
shapiro.test(env$Shannon) #fails hard
leveneTest(env$Shannon~env$Species) #passes


env$Richness <- specnumber(PP_PA)
mean(env$Richness);sd(env$Richness)
# [1] 1.247788
# [1] 0.4915532
shapiro.test(env$Richness) #fails hard
leveneTest(env$Richness~env$Species) #passes

aggregate(.~Species, data=env, mean)
#   Species  Harvest Type     Tree     Site Site_Cat    Shannon Richness
# 1    ABPR 3.000000    2 2.500000 64.00000        1 0.00000000 1.000000
# 2    ALVI 2.690476    1 1.809524 33.42857        1 0.24755256 1.357143
# 3    POTR 2.200000    1 1.400000 38.80000        1 0.20794415 1.400000
# 4    PSME 1.857143    2 1.928571 54.71429        1 0.09902103 1.142857
# 5    SASI 1.736842    1 1.973684 33.52632        1 0.10944429 1.157895
# 6    TSHE 2.142857    2 1.571429 46.00000        1 0.09902103 1.142857


PPdiv_sp <- aov(env$Shannon~env$Species,var.equal=T); summary(PPdiv_sp)
TukeyHSD(PPdiv_sp)
kruskal.test(env$Shannon,env$Species)
#no difs for anything by harvest or tpe either
PPrich_sp <- aov(env$Richness~env$Species,var.equal=T); summary(PPrich_sp)
TukeyHSD(PPrich_sp)
kruskal.test(env$Richness,env$Species)
#no dif for anything by Harvest or by Type either
```


```{r div_SA}
SA_env$Shannon <- diversity(SA_com)
mean(SA_env$Shannon);sd(SA_env$Shannon)
# [1] 0.1819511
# [1] 0.306904


SA_env$Richness <- specnumber(SA_com)
mean(SA_env$Richness);sd(SA_env$Richness)
# [1] 1.2625
# [1] 0.4427689


div_sp <- aov(SA_env$Shannon~SA_env$Species); summary(div_sp)
#no dif p=0.0892
shapiro.test(SA_env$Shannon) #fails
leveneTest(SA_env$Shannon~SA_env$Species) #fails, but just barely, p=0.04
kruskal.test(SA_env$Shannon,SA_env$Species)
# 	Kruskal-Wallis rank sum test
# 
# data:  SA_env$Shannon and SA_env$Species
# Kruskal-Wallis chi-squared = 4.04, df = 1, p-value = 0.04443


rich_sp <- aov(SA_env$Richness~SA_env$Species); summary(rich_sp)
shapiro.test(SA_env$Richness) #fails
leveneTest(SA_env$Richness~SA_env$Species) #fails but just barely, p=0.04
kruskal.test(SA_env$Richness,SA_env$Species)
# 	Kruskal-Wallis rank sum test
# 
# data:  SA_env$Shannon and SA_env$Species
# Kruskal-Wallis chi-squared = 4.04, df = 1, p-value = 0.04443


SA_env$Shannon <- diversity(SA_com)
SA_env$Richness <- specnumber(SA_com)

wilcox.test(SA_env$Shannon~SA_env$Species)
# 	Wilcoxon rank sum test with continuity correction
# 
# data:  SA_env$Shannon by SA_env$Species
# W = 957, p-value = 0.04511
# alternative hypothesis: true location shift is not equal to 0
# 
# Warning message:
# In wilcox.test.default(x = c(0, 0.693147180559945, 0.693147180559945,  :
#   cannot compute exact p-value with ties
# > wilcox.test(SA_env$Shannon~SA_env$Species,correct=F)
# 
# 	Wilcoxon rank sum test
# 
# data:  SA_env$Shannon by SA_env$Species
# W = 957, p-value = 0.04443
# alternative hypothesis: true location shift is not equal to 0
# 
# Warning message:
# In wilcox.test.default(x = c(0, 0.693147180559945, 0.693147180559945,  :
#   cannot compute exact p-value with ties

#warnings because n<50

wilcox.test(SA_env$Richness~SA_env$Species)
# 	Wilcoxon rank sum test with continuity correction
# 
# data:  SA_env$Richness by SA_env$Species
# W = 957, p-value = 0.04511
# alternative hypothesis: true location shift is not equal to 0
# 
# Warning message:
# In wilcox.test.default(x = c(1L, 2L, 2L, 1L, 1L, 2L, 2L, 1L, 1L,  :
#   cannot compute exact p-value with ties
# > wilcox.test(SA_env$Richness~SA_env$Species,correct=F)
# 
# 	Wilcoxon rank sum test
# 
# data:  SA_env$Richness by SA_env$Species
# W = 957, p-value = 0.04443
# alternative hypothesis: true location shift is not equal to 0
# 
# Warning message:
# In wilcox.test.default(x = c(1L, 2L, 2L, 1L, 1L, 2L, 2L, 1L, 1L,  :
#   cannot compute exact p-value with ties


#decid com
D_env$Shannon <- diversity(D_com)
mean(D_env$Shannon);sd(D_env$Shannon)
# [1] 0.1848392
# [1] 0.3252808


D_env$Richness <- specnumber(D_com)
mean(D_env$Richness);sd(D_env$Richness)
# [1] 1.277778
# [1] 0.5198915


div_sp <- aov(D_env$Shannon~D_env$Species); summary(div_sp)
#no dif p=0.162
shapiro.test(D_env$Shannon) #fails
leveneTest(D_env$Shannon~D_env$Species) #passes
kruskal.test(D_env$Shannon,D_env$Species)
# 	Kruskal-Wallis rank sum test
# 
# data:  D_env$Shannon and D_env$Species
# Kruskal-Wallis chi-squared = 4.0717, df = 2, p-value = 0.1306


rich_sp <- aov(D_env$Richness~D_env$Species); summary(rich_sp)
#               Df Sum Sq Mean Sq F value Pr(>F)
# D_env$Species  2   0.96  0.4800   1.808   0.17
# Residuals     87  23.09  0.2655       
shapiro.test(D_env$Richness) #fails
leveneTest(D_env$Richness~D_env$Species) #passes
kruskal.test(D_env$Richness,D_env$Species)
# 
# 	Kruskal-Wallis rank sum test
# 
# data:  D_env$Richness and D_env$Species
# Kruskal-Wallis chi-squared = 4.0717, df = 2, p-value = 0.1306


```


```{r indicsp_PP}
##indicator species analysis
PP_sp<-multipatt(PP_PA,env$Species,func="IndVal.g",permutations=999)
PP_sp
#       s.ABPR s.ALVI s.POTR s.PSME s.SASI s.TSHE index      stat p.value
# Otu13      0      0      1      0      0      0     3 0.4204095   0.065
# Otu16      0      0      1      0      0      0     3 0.4472136   0.040**
# Otu25      0      0      0      0      0      1     6 0.3779645   0.080


PP_type<-multipatt(PP_PA,env$Type,func="IndVal.g",permutations=999)
PP_type
#       s.deciduous s.evergreen index      stat p.value
# Otu11           0           1     2 0.2948839   0.045**
# Otu12           0           1     2 0.3466928   0.035**


PP_harv<-multipatt(PP_PA,env$Harvest,func="IndVal.g",permutations=999)
PP_harv
#       s.August s.July s.September index      stat p.value
# Otu04        0      0           1     3 0.3566223   0.045**
# Otu23        1      0           0     1 0.2581989   0.080
```


```{r indicsp_SA}
SA_sp<-multipatt(SA_com,SA_env$Species,func="IndVal.g",permutations=999)
SA_sp
#       s.ALVI s.SASI index      stat p.value
# Otu03      0      1     2 0.4588315   0.005**
# Otu04      1      0     1 0.4629100   0.010**
# Otu06      0      1     2 0.3244428   0.035**
# Otu07      1      0     1 0.3450328   0.065

SA_harv<-multipatt(SA_com,SA_env$Harvest,func="IndVal.g",permutations=999)
SA_harv
#       s.August s.July s.September index      stat p.value
# Otu03        1      1           0     4 0.3870129   0.075
# Otu04        0      0           1     3 0.4271545   0.030*


#decid indic sp
D_sp <- multipatt(D_com,D_env$Species,func="IndVal.g", permutations = 999)
D_sp
#       s.ALVI s.POTR s.SASI index      stat p.value
# Otu03      0      1      1     6 0.4330127   0.025
# Otu04      1      0      0     1 0.4629100   0.020
# Otu13      0      1      0     2 0.4204095   0.035
# Otu16      0      1      0     2 0.4472136   0.010


D_harv <- multipatt(D_com,D_env$Harvest,func="IndVal.g", permutations = 999)
D_harv
#       s.August s.July s.September index      stat p.value
# Otu04        0      0           1     3 0.3993350   0.025
```


```{r phyloseq_figs}
taxa <- read.csv("adj_365.csv",header=T, row.names=1, stringsAsFactors=T)

fungi <- t(as.matrix(PP_PA))
taxa <- as.matrix(taxa)
env

OTU <- otu_table(fungi,taxa_are_rows = T)
TAX <- tax_table(taxa)

physeq = phyloseq(OTU,TAX)

envdata <- sample_data(env)
sample_names(physeq)
sample_names(envdata)

merged = merge_phyloseq(physeq,envdata)
relabund <- transform_sample_counts(merged,function(x)x/sum(x))

relabund_otu <- otu_table(relabund, taxa_are_rows = FALSE)
write.csv(relabund_otu, file = 'relabund_otu3.csv')
rel <- t(relabund_otu)
write.csv(rel, file = 'relabund_otu_t3.csv')

rel_harvest <- read.csv("rel_h_adj.csv")
rel_species <- read.csv("rel_sp_adj.csv")
rel_type <- read.csv("rel_t_adj.csv")

palette_56 <- c("#b2d200",
"#787362",
"#81ef1e",
"#3b5a3f",
"#f2ff31",
"#58542d",
"#01e959",
"#bbb098",
"#1ac900",
"#889a8b",
"#98e700",
"#597b61",
"#afe500",
"#a9bead",
"#9fff50",
"#7c6c38",
"#b7ff60",
"#106333",
"#e4d700",
"#528f63",
"#f3cb26",
"#00bd60",
"#ffe04b",
"#235f0a",
"#fbffda",
"#01ba2b",
"#fff0b7",
"#259800",
"#dcffd3",
"#226d00",
"#ffff74",
"#4d5816",
"#90ff82",
"#615c00",
"#7aff9e",
"#8b7800",
"#b3ff99",
"#4a5a00",
"#e7ff84",
"#2c9956",
"#d6b439",
"#019838",
"#ffe392",
"#4ca000",
"#dac387",
"#6fbf00",
"#a4915c",
"#a5b800",
"#84be92",
"#a09d00",
"#66d187",
"#b59830",
"#ddffb5",
"#3c8100",
"#e9ff9c",
"#758200")

species_order <- c("ALVI","POTR","SASI","ABPR","PSME","TSHE")

g <- ggplot(rel_species, aes(x=factor(Species,level=species_order),Abundance)); g
 p<- g + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="bottom",axis.title.x = element_text(size=24),axis.title.y = element_text(size=24),axis.text.y = element_text(size=20),axis.text.x = element_blank()) + scale_fill_manual(values=palette_56, guide = guide_legend(nrow=4,ncol=10))+labs(x="Species")+geom_vline(xintercept = 3.5, linetype="solid", 
                color = "black", size=1) + annotate("text",x=c(1,2,3,4,5,6),y=c(0.95,0.95,0.95,0.95,0.95,0.95),label=c("ALVI","POTR","SASI","ABPR","PSME","TSHE"),size=6)
 p
 
 p<- g + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="bottom",axis.title.x = element_text(size=24),axis.title.y = element_text(size=24),axis.text.y = element_text(size=20),axis.text.x = element_text(size=20)) + scale_fill_manual(values=palette_56, guide = guide_legend(nrow=4,ncol=10))+labs(x="Species")+geom_vline(xintercept = 3.5, linetype="solid", 
                color = "black", size=1) #+ annotate("text",x=c(1,2,3,4,5,6),y=c(0.95,0.95,0.95,0.95,0.95,0.95),label=c("ALVI","POTR","SASI","ABPR","PSME","TSHE"),size=6)
 p

jpeg(filename = "abundance_OTU_adj.jpeg",width=12, height=10, units="in", res = 300)
 p
dev.off()


 bp<- g + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="bottom",axis.title.x = element_text(size=24),axis.title.y = element_text(size=24),axis.text.y = element_text(size=20),axis.text.x = element_text(size=20)) + scale_fill_grey(start=0,end=0.9, guide = guide_legend(nrow=4,ncol=10))+labs(x="Species")+geom_vline(xintercept = 3.5, linetype="solid", 
                color = "black", size=1) #+ annotate("text",x=c(1,2,3,4,5,6),y=c(0.95,0.95,0.95,0.95,0.95,0.95),label=c("ALVI","POTR","SASI","ABPR","PSME","TSHE"),size=6)
 bp

#make black and white version....
b<- g + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="right",axis.title.x = element_text(size=24),axis.title.y = element_text(size=24),axis.text.y = element_text(size=20),axis.text.x = element_text(size=20)) + scale_fill_grey(start = 0, end = 0.9, guide = guide_legend(nrow=28,ncol=2))+labs(x="Species")+geom_vline(xintercept = 3.5, linetype="solid", 
                color = "black", size=1)
jpeg(filename = "abundance_OTU_adj_bw.jpeg",width=12, height=10, units="in", res = 300)
bp
dev.off()

 
level_order <- c('July', 'August', 'September')
 
 m <- ggplot(rel_harvest, aes(x = factor(Harvest, level = level_order),Abundance)); m
 n<- m + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="bottom",axis.title.x = element_text(size=24),axis.title.y = element_text(size=24),axis.text.y = element_text(size=20),axis.text.x = element_text(size=20)) + scale_fill_manual(values=palette_56, guide = guide_legend(nrow=4,ncol=10))+labs(x="")
 n

jpeg(filename = "harvest_OTU_adj.jpeg",width=12, height=10, units="in", res = 300)
 n 
dev.off() 


#new version
 timebw<- m + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="bottom",axis.title.x = element_text(size=24),axis.title.y = element_text(size=24),axis.text.y = element_text(size=20),axis.text.x = element_text(size=20)) + scale_fill_grey(start=0,end=0.9, guide = guide_legend(nrow=4,ncol=10))+labs(x="")
 timebw
 

#bw version
nbw <- m + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="right",axis.title.x = element_text(size=24),axis.title.y = element_text(size=24),axis.text.y = element_text(size=20),axis.text.x = element_text(size=20)) + scale_fill_grey(start=0,end=0.9, guide = guide_legend(nrow=28,ncol=2))+labs(x="")

jpeg(filename = "harvest_OTU_adj_bw.jpeg",width=12, height=10, units="in", res = 300)
nbw
dev.off()


c <- ggplot(rel_type, aes(Type,Abundance)); c
t<- c + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="bottom",legend.direction="horizontal",axis.title.x = element_text(size=24),axis.title.y = element_blank(),axis.text.y=element_blank(),axis.text.x = element_text(size=20)) + scale_fill_manual(values=palette_56, guide = guide_legend(nrow=4,ncol=10))+labs(x="")
t

#new one
typebw<- c + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="bottom",legend.direction="horizontal",axis.title.x = element_text(size=24),axis.title.y = element_blank(),axis.text.y=element_blank(),axis.text.x = element_text(size=20)) + scale_fill_grey(start=0,end=0.9, guide = guide_legend(nrow=4,ncol=10))+labs(x="")
typebw

jpeg(filename = "type_OTU_adj.jpeg",width=12, height=10, units="in", res = 300)
t
dev.off()

#bw version
tbw<- c + geom_bar(stat="identity",aes(fill=OTU),color="black")+
    theme_bw() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(), legend.position="right",axis.title.x = element_text(size=24),axis.title.y = element_text(size=24),axis.text.y = element_text(size=20),axis.text.x = element_text(size=20)) + scale_fill_grey(start=0,end=0.9, guide = guide_legend(nrow=28,ncol=2))

jpeg(filename = "type_OTU_adj_bw.jpeg",width=12, height=10, units="in", res = 300)
tbw
dev.off()

# #theme(legend.key.width=unit(1,"cm"),
# #(merged, "Species", fill = "OTU") + theme(legend.position="bottom") + scale_y_continuous(limits = c(0,1.0), expand = c(0, 0)) + scale_fill_manual(values=c("#b2d200",
# "#787362",
# "#81ef1e",
# "#3b5a3f",
# "#f2ff31",
# "#58542d",
# "#01e959",
# "#bbb098",
# "#1ac900",
# "#889a8b",
# "#98e700",
# "#597b61",
# "#afe500",
# "#a9bead",
# "#9fff50",
# "#7c6c38",
# "#b7ff60",
# "#106333",
# "#e4d700",
# "#528f63",
# "#f3cb26",
# "#00bd60",
# "#ffe04b",
# "#235f0a",
# "#fbffda",
# "#01ba2b",
# "#fff0b7",
# "#259800",
# "#dcffd3",
# "#226d00",
# "#ffff74",
# "#4d5816",
# "#90ff82",
# "#615c00",
# "#7aff9e",
# "#8b7800",
# "#b3ff99",
# "#4a5a00",
# "#e7ff84",
# "#2c9956",
# "#d6b439",
# "#019838",
# "#ffe392",
# "#4ca000",
# "#dac387",
# "#6fbf00",
# "#a4915c",
# "#a5b800",
# "#84be92",
# "#a09d00",
# "#66d187",
# "#b59830",
# "#ddffb5",
# "#3c8100",
# "#e9ff9c",
# "#758200"))
# g

###bargraph of of distribution of cultures?

library(ggpubr)
col_panel <- ggarrange(n, t, 
          labels = c("A", "B"),
          common.legend = T,legend="bottom",
          ncol=2,nrow=1,
          font.label = list(size = 20, color = "black"),
          hjust =c(-6,-1))

col_panel

jpeg(filename = "col-panel.jpeg",width=12, height=10, units="in", res = 300)
col_panel
dev.off()

bw_panel <- ggarrange(timebw,typebw, 
          labels = c("A", "B"),
          common.legend = T,legend="bottom",
          ncol=2,nrow=1,
          font.label = list(size = 20, color = "black"),
          hjust =c(-6,-1))
bw_panel

jpeg(filename = "bw-panel.jpeg",width=12, height=10, units="in", res = 300)
bw_panel
dev.off()

```

maps!
```{r}
gps <- read.csv("C:/Users/emyle/Dropbox/School/Data/Projects/MSH/GPS points/formap.csv")
head(gps)

#for satellite imagery:
#bullshit API key is required now
#made an account but had to set up billing

library(ggmap)
library(ggrepel)
MSHPP <- c(lon=mean(gps$Longitude), lat=mean(gps$Latitude))

lats<-c(46.239,46.265)
lons<-c(-122.20,-122.15)
bb<-make_bbox(lon=lons,lat=lats,f=0.05)

msh_bb <- c(left = -122.20,
            bottom = 46.239,
            right = -122.15,
            top = 46.265)

mean(46.239,46.265)


MSHPP_center <- c(lon=-122.173, lat=46.249)

register_google(key="AIzaSyCankhCsiF7SZ71OQJX-qW1XHbPq6B080g")
zoom15 <- get_map(location=MSHPP_center,zoom=15, maptype="satellite")
zoom14 <- get_map(msh_bb,zoom=15, maptype="satellite")

ggmap(zoom15)
ggmap(zoom14)

bm <- get_map(bbox=msh_bb,zoom=14,maptype = "satellite")
ggmap(bm)

anchor<-c(x=-122.15,y=46.245)

bigmap<-ggmap(zoom15) + geom_point(data=gps,aes(Longitude,Latitude),alpha = 1, fill = "white", color="black",pch = 21, size = 4,stroke=2) + labs(x = 'Longitude', y = 'Latitude') + geom_label_repel(data = gps, aes(Longitude, Latitude, label = Cluster), fill = "white") + scalebar(y.min = 46.25374, y.max = 46.2543, x.min = -122.185, x.max = -122.178, dist=250,dist_unit = "m",transform=T,model="WGS84",st.dist = 0.5, height=0.5) #+ annotate("segment", x = -122.161, xend = -122.1598, y = 46.257, yend = 46.258, colour = "white", size=2, alpha=1, arrow=arrow()) #+ annotate("text",label="Spirit Lake", x=-122.1599,y=46.2569)
  
  #scalebar(y.min = 46.232, y.max = 46.233, x.min = -122.20, x.max = -122.18, dist=0.5,dist_unit = "km",transform=T,model="WGS84",st.dist = 0.5, height=0.5) #+ north(location="bottomright",symbol=10,x.min=-122.15,x.max=-122.145,y.min=46.24,y.max=46.245,anchor=T)

#jpeg(filename = "overview_zoom13.jpeg",units="in",width =7, height=7, res=300) #saves graph output
bigmap
#dev.off()


install.packages("tigris")
install.packages("cowplot")
library(tigris)
library(ggplot2)
library(cowplot)

install.packages("tmap")
install.packages(c("sf","sp"))
install.packages("ggsn")
library(ggsn)
library(tmap)
library(sf)
library(sp)

us_states_20m = states(cb=T,resolution = "20m")
continental <- subset(us_states_20m,!NAME %in% c(
                     "United States Virgin Islands",
                     "Commonwealth of the Northern Mariana Islands",
                     "Guam",
                     "American Samoa",
                     "Puerto Rico",
                     "Alaska",
                     "Hawaii"
                   ))

us_states = subset(us_states, 
                   !NAME %in% c(
                     "United States Virgin Islands",
                     "Commonwealth of the Northern Mariana Islands",
                     "Guam",
                     "American Samoa",
                     "Puerto Rico",
                     "Alaska",
                     "Hawaii"
                   ))


data(rivers)

str(us_states)

US <- ggplot() +
  geom_sf(data = continental, fill = "white") +
  coord_sf(crs = st_crs(continental), datum = 4326) +
  geom_point(data=MSHdf,x=lon,y=lat,size=2) +
  theme(line = element_blank(),
        text = element_blank(),
        title = element_blank(),
        panel.background = element_rect(fill = "white", colour = "grey50"))

US

#WGS84 is 4326

continental_t <- st_transform(continental,crs=2163)

sitept <- st_transform(MSHdf,crs=4326)

site <- c("PP")
lat <- c(46.24743)
lon<-c(-122.17031)
MSHdf <- data.frame(lon,lat)
MSHdf

df<- structure(list(longitude = c(-122.17031), latitude = c(46.24743)), .Names = c("longitude", 
"latitude"), class = "data.frame", row.names = c(NA, -1L))

df

xy <- df[,c(1,2)]

spdf <- SpatialPointsDataFrame(coords = xy, data = df,
                               proj4string = CRS("+init=epsg:2163"))

gg_inset_map2 = ggdraw() +
  draw_plot(bigmap) +
  draw_plot(US, x = 0.1, y = 0.7, width = 0.35, height = 0.35)

dev.off()
jpeg(filename = "overview_zoom15.jpeg",units="in",width =7, height=7, res=300) #saves graph output
gg_inset_map2
dev.off()




# grab a center/zoom map and compute its bounding box
gc <- geocode("white house, washington dc")
map <- get_map(gc)
(bb <- attr(zoom14, "bb"))
(bbox <- bb2bbox(bb))

# use the bounding box to get a stamen map
stamMap <- get_stamenmap(bbox)

ggmap(map) +
  geom_point(
    aes(x = lon, y = lat),
    data = gc, colour = "red", size = 3
  )

ggmap(stamMap) +
  geom_point(
    aes(x = lon, y = lat),
    data = gc, colour = "red", size = 3
  )

```
